File: apiwznm/WznmQTcoATitle.cpp
/**
	* \file WznmQTcoATitle.cpp
	* API code for table TblWznmQTcoATitle (implementation)
	* \author Alexander Wirthmueller
	* \date created: 11 Jul 2020
	* \date modified: 11 Jul 2020
	*/

#include "WznmQTcoATitle.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQTcoATitle
 ******************************************************************************/

WznmQTcoATitle::WznmQTcoATitle(
			const uint jnum
			, const string srefX1IxVType
			, const string titX1IxVType
			, const string stubX2RefWznmMLocale
			, const string Title
		) {
	this->jnum = jnum;
	this->srefX1IxVType = srefX1IxVType;
	this->titX1IxVType = titX1IxVType;
	this->stubX2RefWznmMLocale = stubX2RefWznmMLocale;
	this->Title = Title;
};

bool WznmQTcoATitle::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQTcoATitle");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "srefX1IxVType", "typ", srefX1IxVType);
		extractStringUclc(docctx, basexpath, "titX1IxVType", "typ2", titX1IxVType);
		extractStringUclc(docctx, basexpath, "stubX2RefWznmMLocale", "loc", stubX2RefWznmMLocale);
		extractStringUclc(docctx, basexpath, "Title", "tit", Title);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQTcoATitle
 ******************************************************************************/

ListWznmQTcoATitle::ListWznmQTcoATitle() {
};

ListWznmQTcoATitle::ListWznmQTcoATitle(
			const ListWznmQTcoATitle& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQTcoATitle(*(src.nodes[i]));
};

ListWznmQTcoATitle& ListWznmQTcoATitle::operator=(
			const ListWznmQTcoATitle& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQTcoATitle(*(src.nodes[i]));

	return *this;
};

ListWznmQTcoATitle::~ListWznmQTcoATitle() {
	clear();
};

void ListWznmQTcoATitle::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQTcoATitle::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQTcoATitle* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQTcoATitle");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQTcoATitle", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQTcoATitle(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQTcoATitle[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japiwznm/WznmQTcoATitle.java
/**
  * \file WznmQTcoATitle.h
  * Java API code for record of table TblWznmQTcoATitle
  * \author Alexander Wirthmueller
  * \date created: 11 Jul 2020
  * \date modified: 11 Jul 2020
  */

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQTcoATitle {
	
	public WznmQTcoATitle(
				int jnum
				, String srefX1IxVType
				, String titX1IxVType
				, String stubX2RefWznmMLocale
				, String Title
			) {
		this.jnum = jnum;
		this.srefX1IxVType = srefX1IxVType;
		this.titX1IxVType = titX1IxVType;
		this.stubX2RefWznmMLocale = stubX2RefWznmMLocale;
		this.Title = Title;
	};

	public int jnum;
	public String srefX1IxVType;
	public String titX1IxVType;
	public String stubX2RefWznmMLocale;
	public String Title;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQTcoATitle");

		if (Xmlio.checkXPath(doc, basexpath)) {
			srefX1IxVType = Xmlio.extractStringUclc(doc, basexpath, "srefX1IxVType", "typ", null, 0);
			titX1IxVType = Xmlio.extractStringUclc(doc, basexpath, "titX1IxVType", "typ2", null, 0);
			stubX2RefWznmMLocale = Xmlio.extractStringUclc(doc, basexpath, "stubX2RefWznmMLocale", "loc", null, 0);
			Title = Xmlio.extractStringUclc(doc, basexpath, "Title", "tit", null, 0);
			
			return true;
		};

		return false;
	};
};

